pkg/entities/ie.go

```diff
@@ -388,3 +388,158 @@ func EncodeToIEDataType(dataType IEDataType, val interface{}) ([]byte, error) {
 	}
 	return nil, fmt.Errorf("API supports only valid information elements with datatypes given in RFC7011")
 }
+
+// EncodeToBuff is to encode data to specific type to the buff
+func EncodeToBuff(dataType IEDataType, val interface{}, buffer []byte, index int) error {
```

Comment:
This is good. Just worried about the scenario if the buffer does not have enough length, where it will cause panic. Maybe have a check before calling this function?

Reply:
Makes sense. Added the check and make this method private since it will not be used elsewhere